crates/biome_graphql_parser/src/lexer/mod.rs

```diff
@@ -0,0 +1,720 @@
+                if self.current_byte() == Some(b'"') && self.byte_at(1) == Some(b'"') {
+                    self.advance(2);
+                    LexStringState::Terminated
+                } else {
+                    state
+                }
+            }
+            _ => panic!("String terminated"),
+        }
+    }
+
+    fn consume_escape_sequence_in_string(
+        &mut self,
+        chr: u8,
+        state: LexStringState,
+    ) -> (LexStringState, Option<ParseDiagnostic>) {
+        assert_eq!(chr, b'\\');
+        let escape_start = self.text_position();
+        self.advance(1);
+        match state {
+            // '\t' etc
+            LexStringState::InString => match self.current_byte() {
+                Some(b'"' | b'\\' | b'/' | b'b' | b'f' | b'n' | b'r' | b't') => {
+                    self.advance(1);
+                    (state, None)
+                }
+
+                Some(b'u') => match self.consume_unicode_escape() {
+                    Err(diagnostic) => (state, Some(diagnostic)),
+                    Ok(_) => (state, None),
+                },
+
+                Some(_) => {
+                    let c = self.current_char_unchecked();
+                    let diagnostic = ParseDiagnostic::new(
+                        "Invalid escape sequence",
+                        escape_start..self.text_position() + c.text_len(),
+                    ).with_hint(r#"Valid escape sequences are: `\\`, `\/`, `/"`, `\b\`, `\f`, `\n`, `\r`, `\t` or any unicode escape sequence `\uXXXX` where X is hexedecimal number. "#);
+                    (state, Some(diagnostic))
+                }
+
+                None => (
+                    state,
+                    Some(
+                        ParseDiagnostic::new(
+                            "Expected an escape sequence following a backslash, but found none",
+                            escape_start..self.text_position(),
+                        )
+                        .with_detail(self.text_position()..self.text_position(), "File ends here"),
+                    ),
+                ),
+            },
+            // '\"""'
+            LexStringState::InBlockString => {
+                if self.current_byte() == Some(b'"')
+                    && self.byte_at(1) == Some(b'"')
+                    && self.byte_at(2) == Some(b'"')
+                {
+                    self.advance(3);
+                    (state, None)
+                } else {
+                    let c = self.current_char_unchecked();
+                    let diagnostic = ParseDiagnostic::new(
+                        "Invalid escape sequence",
+                        escape_start..self.text_position() + c.text_len(),
+                    )
+                    .with_hint(r#"For block string the only valid escape sequences is `\"""`. "#);
+                    (state, Some(diagnostic))
+                }
+            }
+            _ => panic!("String uninitialized/terminated"),
```

Comment:
We don't want to panic in production. Consider using a safe approach, maybe by emitting a diagnostic and/or using `consume_unexpecated_character`